src/components/organisms/CreateFileFolderModal/CreateFileFolderModal.tsx

```diff
@@ -0,0 +1,124 @@
+
+import {CreateFileFolderCallback, checkIfEntityExists, createFile, createFolder} from '@utils/files';
+import {useFocus} from '@utils/hooks';
+
+const prohibitedFirstSymbols = ['/', '\\'];
+
+const CreateFileFolderModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const uiState = useAppSelector(state => state.ui.createFileFolderModal);
+
+  const [createFileFolderForm] = useForm();
+
+  const [inputRef, focus] = useFocus<any>();
+
+  const onFinish = (values: {fileFolderName: string}) => {
+    const {fileFolderName} = values;
+
+    if (uiState.type === 'folder') {
+      createFolder(uiState.rootDir, fileFolderName, onCreate);
+    } else {
+      createFile(uiState.rootDir, fileFolderName, onCreate);
+    }
+  };
+
+  const onCreate = (args: CreateFileFolderCallback) => {
+    const {fileFolderName, err} = args;
+
+    if (err) {
+      dispatch(
+        setAlert({
+          title: 'Creating failed',
+          message: `Something went wrong during the creation of a ${uiState.type}`,
+          type: AlertEnum.Error,
+        })
+      );
+    } else {
+      dispatch(
+        setAlert({
+          title: `Successfully created a ${uiState.type}`,
+          message: `You have successfully created the "${fileFolderName}" ${uiState.type}`,
+          type: AlertEnum.Success,
+        })
+      );
+
+      dispatch(closeCreateFileFolderModal());
+    }
+  };
+
+  useEffect(() => {
+    if (uiState.isOpen) {
+      focus();
+    } else {
+      createFileFolderForm.resetFields();
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [uiState]);
+
+  if (!uiState.isOpen) {
+    return null;
+  }
+
+  return (
+    <Modal
+      title={`Create ${uiState.type}`}
+      visible={uiState.isOpen}
+      onCancel={() => {
+        dispatch(closeCreateFileFolderModal());
+      }}
+      okText="Submit"
+      onOk={() => {
+        createFileFolderForm.submit();
+      }}
+    >
+      <Form layout="vertical" form={createFileFolderForm} onFinish={onFinish}>
+        <Form.Item
+          label={`${uiState.type === 'file' ? 'File' : 'Folder'} name`}
+          name="fileFolderName"
+          rules={[
+            ({getFieldValue}) => ({
+              validator: () => {
+                return new Promise((resolve: (value?: any) => void, reject) => {
+                  const fileFolderNameValue: string = getFieldValue('fileFolderName').toLowerCase();
+
+                  if (!fileFolderNameValue) {
+                    reject(new Error("This field can't be empty"));
+                  }
+
+                  if (prohibitedFirstSymbols.some(symbol => symbol === fileFolderNameValue[0])) {
+                    reject(
+                      new Error(`${uiState.type} name can't start with prohibited symbol - "${fileFolderNameValue[0]}"`)
+                    );
+                  }
+
+                  if (checkIfEntityExists(path.join(uiState.rootDir, path.sep, fileFolderNameValue))) {
```

Comment:
I don't think `path.sep` is needed when using `path.join`.
Also, the `checkIfEntityExists` function name is not very intuitive, maybe it should be renamed to `doesPathExist` ?